src/Features/LanguageServer/Protocol/Handler/References/FindReferencesHandler.cs

```diff
@@ -1,65 +1,57 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 using System.Collections.Generic;
+using System.Composition;
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Classification;
 using Microsoft.CodeAnalysis.Editor.FindUsages;
 using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
-using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
 using Microsoft.CodeAnalysis.FindUsages;
-using Microsoft.CodeAnalysis.LanguageServer;
 using Microsoft.CodeAnalysis.PooledObjects;
-using Microsoft.VisualStudio.LanguageServer.Protocol;
-using Microsoft.VisualStudio.LiveShare.LanguageServices;
-using Microsoft.VisualStudio.Shell;
 using Microsoft.VisualStudio.Text.Adornments;
 using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
 
-namespace Microsoft.VisualStudio.LanguageServices.LiveShare
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler
 {
-    internal class FindAllReferencesHandler : ILspRequestHandler<LSP.ReferenceParams, object[], Solution>
+    [Shared]
+    [ExportLspMethod(LSP.Methods.TextDocumentReferencesName)]
+    internal class FindAllReferencesHandler : IRequestHandler<LSP.ReferenceParams, object[]>
     {
-        private readonly IThreadingContext _threadingContext;
-
-        public FindAllReferencesHandler(IThreadingContext threadingContext)
-        {
-            _threadingContext = threadingContext;
-        }
-
-        public async Task<object[]> HandleAsync(LSP.ReferenceParams request, RequestContext<Solution> requestContext, CancellationToken cancellationToken)
+        /// <summary>
+        /// Keep thread context by default - currently FAR requires the UI thread to get third party references.
+        /// TODO - this should not require the UI thread...
+        /// </summary>
+        /// <param name="solution"></param>
+        /// <param name="request"></param>
+        /// <param name="clientCapabilities"></param>
+        /// <param name="cancellationToken"></param>
+        /// <param name="keepThreadContext"></param>
+        /// <returns></returns>
```

Comment:
remove non-valuable doc comemnts (i.e. the empty params/returns).  or fill them out.